src/pymoca/ast.py

```diff
@@ -895,6 +901,18 @@ def remove_initial_equation(self, e: Equation) -> None:
         """
         self.initial_equations.remove(e)
 
+    def extend(self, other: "Class") -> None:
+        self._extend(other)
+        self.update_parent_refs()
```

Comment:
I'm not sure how expensive it is to do this recursively like this. Only the top level really needs their parent updated right?

Reply:
I think this is fixed with 77c9db8 "Tree: Fix partial instantiation of parents until root".